File: api/Models/Discount.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ByzmoApi.Models
{
    public class Discount
    {
        public int Id { get; set; }
        public string Code { get; set; }
        public string Description { get; set; }
        public int AmountTypeId { get; set; }
        public decimal Amount { get; set; }
        public DateTimeOffset StartDate { get; set; }
        public DateTimeOffset EndDate { get; set; }
        public bool IsActive { get; set; }
        public bool IsOneTimeUse { get; set; }
        public bool IsClaimed { get; set; }

        
    }
}

File: web/src/app/classes/discount.ts
export class Discount {
    id: number;
    code: string;
    description: string;
    amountTypeId: number;
    amount: number;
    startDate: Date;
    endDate: Date;
    isActive: boolean;
    isOneTimeUse: boolean;
    constructor() {
        this.amount = 0;
        this.amountTypeId = 0;
    }
}
